### DISCUSSION TITLE
Recommended shaders for wide variety of models?

### DISCUSSION THREAD

**Discussion by ryan-motive** [atteneder/glTFast#563] [Q&A]
We're considering our options for runtime GLTF import. Speed is important so we're quite eager to use glTFast. I will admit to not being extremely knowledgeable about rendering and shaders. Is there a good subset of shader variants that would cover most cases? For example, could we choose a list of say 10% of all possible variants that would cover 90% of the common cases? It would be fine on our side to specify what shaders are supported by default and to give guidelines to devs for how to extend the list if needed.

**Comment by atteneder** [comment]
@ryan-motive That's a great suggestion!

What I do on my test projects is loading all of the sample assets (so essentially a representative set of shader variants), capture a shader variant collection and using it.

It would take some more research on actual projects/real world data to get a good understanding of most used variants.

A different solution would be to load shader variants on-demand. At this stage this is just an idea and lacks a feasibility study.
